client/controller/api/apps.go

```diff
@@ -6,7 +6,7 @@ type App struct {
 	ID      string `json:"id"`
 	Owner   string `json:"owner"`
 	Updated string `json:"updated"`
-	URL     string `json:"url"`
+	URL     string
```

Comment:
Yea, keep the struct for now. You're decoding JSON into all of the struct except for `URL`. The `json:"-"` is a [struct tag for the JSON encoder](https://golang.org/pkg/encoding/json/#Marshal) (and decoder) that tells the marshaller/unmarshaler to ignore `URL` completely.

So, in short, `json.Marshal` and `json.Unmarshal` will ignore `URL` completely, and you can set it afterward.